src/main/java/com/enterprise/eventmanagementsystem/controller/EventController.java

```diff
@@ -38,7 +38,7 @@
     @GetMapping("/")
     public String viewHomePage(Model model) throws Exception {
         model.addAttribute("listEvents", eventService.fetchAll());
-        model.addAttribute("eventsSearchableData", generateEventsSearchableData());
+        model.addAttribute("eventsSearchableData", eventService.generateEventsSearchableData());
```

Comment:
You had a method in your controller that generated a list of searchable data. I moved this method to the service layer, as this method seemed more in place with the rest of the business logic.